src/test/java/org/ow2/proactive/catalog/util/RawObjectResponseCreatorTest.java

```diff
@@ -0,0 +1,130 @@
+                                                          new byte[0]);
+        ResponseEntity responseEntity = rawObjectResponseCreator.createRawObjectResponse(rawObject);
+        assertThat(responseEntity).isNotNull();
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getHeaders().getContentType()).isEqualTo(MediaType.APPLICATION_XML);
+        assertThat(responseEntity.getHeaders().containsKey(HttpHeaders.CONTENT_DISPOSITION)).isTrue();
+        assertThat(responseEntity.getHeaders()
+                                 .getFirst(HttpHeaders.CONTENT_DISPOSITION)).isEqualTo(contentDispositionFileName);
+    }
+
+    @Test
+    public void testCreateRawObjectResponseWorkflowKindWithExtension() {
+        String objectName = "object name.xml";
+        String contentDispositionFileName = "attachment; filename=\"" + objectName + "\"";
+        CatalogRawObject rawObject = new CatalogRawObject(1L,
+                                                          objectName,
+                                                          "workflow",
+                                                          "application/xml",
+                                                          1400343L,
+                                                          "commit message",
+                                                          Collections.emptyList(),
+                                                          new byte[0]);
+        ResponseEntity responseEntity = rawObjectResponseCreator.createRawObjectResponse(rawObject);
+        assertThat(responseEntity).isNotNull();
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getHeaders().getContentType()).isEqualTo(MediaType.APPLICATION_XML);
+        assertThat(responseEntity.getHeaders().containsKey(HttpHeaders.CONTENT_DISPOSITION)).isTrue();
+        assertThat(responseEntity.getHeaders()
+                                 .getFirst(HttpHeaders.CONTENT_DISPOSITION)).isEqualTo(contentDispositionFileName);
+    }
+
+    @Test
+    public void testCreateRawObjectResponseWrongContentType() {
+        CatalogRawObject rawObject = new CatalogRawObject(1L,
+                                                          "name",
+                                                          "object",
+                                                          "testContentType",
+                                                          1400343L,
+                                                          "commit message",
+                                                          Collections.emptyList(),
+                                                          new byte[0]);
+        ResponseEntity responseEntity = rawObjectResponseCreator.createRawObjectResponse(rawObject);
+        assertThat(responseEntity).isNotNull();
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
```

Comment:
The wrong content type return an OK response ?

Reply:
exactly. This what we decided long time ago in this issue:
https://github.com/ow2-proactive/catalog/issues/41